### PR TITLE
fix bug in case the no district was selected and add logic for books …

### PR DESCRIPTION
**Pull request by phiele**
…with no locations

+ add logic for books with no locations, so it says "Sorry, but this book is currently not available..."

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -4,18 +4,20 @@ class BooksController < ApplicationController
...
     # get users that have the selected book
     @book_users = @book.users
     # match book users with selected district, e. g. "Neukölln"
     @locations = []
-    @book_users.each do |book_user|
-      @locations += book_user.locations.select { |location| location.district == @search_district }
+    # check if params[:district] was passed
+    if params[:district].nil?
+      @book_users.each do |book_user|
+        @locations += book_user.locations.select { |location| location }
```
Comment (rudygasson) [inline app/controllers/books_controller.rb:14]: This looks odd to me. I think the select statement does nothing here. 
`@locations += book_user.locations` would give the exact same result.

File: app/views/books/show.html.erb [hunk 2]
```diff
@@ -14,19 +14,22 @@
...
+    <% if @locations_uniq.empty? %>
+      <p><em>Sorry, but this book is currently not available...</em></p>
+    <% else %>
+      <% @locations_uniq.each do |location| %>
+        <div class="container mb-3">
+          <h4 class="m-0"><%= location.name %></h4>
+          <%# show users that offer the book at this location %>
+          <p class="m-0">Set up a meeting with:</p>
+          <% location.users.each do |user| %>
+            <% if (user.copies.find_by book_id: params[:id]) %>
```
Comment (rudygasson) [inline app/views/books/show.html.erb:26]: Just as a general reminder: this kind of logic should live in the controller, not in the view. We should at least refactor it later if there is no time now.

**Review by rudygasson — APPROVED** [review]
2 comments in the code, maybe change at least the first point before merging. Cheers

File: app/controllers/books_controller.rb (hunk 1 above)
Comment (phiele) [reply app/controllers/books_controller.rb]: True, I fixed it.

**Pull request merged**